### PR TITLE
[BD-21] Sphinx extension for documenting feature toggles

### PR DESCRIPTION
**Pull request by regisb**
**Description:** We introduce a new Sphinx extension for generating human-readable
documentation of feature toggle annotations. This extension introduces a
`.. featuretoggles::` directive that will list feature toggles for the
configured code base. Annotations are collected on-the-fly using
code_annotations.

**JIRA:** https://openedx.atlassian.net/wiki/spaces/COMM/pages/1596358943

**Dependencies:** related edx-platform PR: https://github.com/edx/edx-platform/pull/24344

**Reviewers:**
- [ ] @robrap 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Concerns:** The related edx-platform PR requires that edx_toggles be pushed to pypi. In the meantime, edx_toggles will be installed in edx-platform from this branch.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @regisb! I've created [BLENDED-499](https://openedx.atlassian.net/browse/BLENDED-499) to keep track of it in Jira.   More details are on the [BD-21](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1596358943/BD-21+Toggles+Settings+Documentation) project page.

When this pull request is ready, tag your edX technical lead.

File: AUTHORS [hunk 1]
```diff
@@ -1 +1,2 @@
 Stuart Young <syoung@edx.org>
+Régis Behmo <regis@behmo.com>
```
Comment (robrap) [inline AUTHORS:2]: I think we should remove this file and remove the bullet in the PR template. :)

File: CHANGELOG.rst [hunk 2]
```diff
@@ -6,26 +6,22 @@ Change Log
...
-~~~~~~~~~~
-* Modified scripts/feature_toggle_report_generator to work based on envs
+[0.3.0] - 2020-07-22
+~~~~~~~~~~~~~~~~~~~~
 
-2020-06-30
-~~~~~~~~~~
-* Added CsvResnderer
+* Add ``featuretoggles`` Sphinx extension
+* Modified scripts/feature_toggle_report_generator to work based on envs
```
Comment (robrap) [inline CHANGELOG.rst:18]: FYI: @jinder1s will soon be merging https://github.com/edx/edx-toggles/pull/40 as well.  Not sure whether or not will will need to those entries as well, or if he'll move this to 0.4.0.

File: edx_toggles/sphinx/extensions/featuretoggles.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    def run(self):
+        toggle_nodes = list(self.iter_nodes())
+        return [nodes.section("", *toggle_nodes, ids=["featuretoggles"])]
+
+    def iter_nodes(self):
+        """
+        Iterate on the docutils nodes generated by this directive.
+        """
+        toggles = find_feature_toggles(self.env.config.featuretoggles_source_path)
+        # import ipdb; ipdb.set_trace()
```
Comment (robrap) [inline edx_toggles/sphinx/extensions/featuretoggles.py:89]: Remove.

**Comment by jinder1s** [comment]
@regisb I just accidentially merged my branch to master rather than squash. The commit history is littered with new commits, to avoid merge conflicts later, I recommend you rebase with master now, rather than later. Especially in case you are making any changes to scripts directory

**Comment by regisb** [comment]
@robrap I updated this PR to move the `feature_toggle_annotations.yaml` file from the repo root to the package folder. This is the only way I know to include the config file in the pip-installed package. I hope this will not break anything elsewhere? As far as I know, this file is only used in this repo: https://github.com/search?q=org%3Aedx+feature_toggle_annotations.yaml&type=Code

With this change, I can now generate the featuretoggle docs in the edx-platform repo from my PR, so this should be ready to merge.

**Comment by regisb** [comment]
@robrap I updated this PR to move the `feature_toggle_annotations.yaml` file from the repo root to the package folder. This is the only way I know to include the config file in the pip-installed package. I hope this will not break anything elsewhere? As far as I know, this file is only used in this repo: https://github.com/search?q=org%3Aedx+feature_toggle_annotations.yaml&type=Code

With this change, I can now generate the featuretoggle docs in the edx-platform repo from my PR, so this should be ready to merge.

**Comment by regisb** [comment]
This PR was made obsolete by the changes made to the code-annotations repository.

**Pull request closed without merging**